js_modules/dagster-ui/packages/ui-core/src/jobs/JobsPageContent.tsx

```diff
@@ -1,93 +1,132 @@
+  useDefinitionTagFilter,
+  useTagsForObjects,
+} from '../ui/Filters/useDefinitionTagFilter';
 import {WorkspaceContext} from '../workspace/WorkspaceContext/WorkspaceContext';
-import {WorkspaceLocationNodeFragment} from '../workspace/WorkspaceContext/types/WorkspaceQueries.types';
+import {
+  WorkspaceLocationNodeFragment,
+  WorkspacePipelineFragment,
+} from '../workspace/WorkspaceContext/types/WorkspaceQueries.types';
 import {buildRepoAddress} from '../workspace/buildRepoAddress';
 import {repoAddressAsHumanString} from '../workspace/repoAddressAsString';
 import {RepoAddress} from '../workspace/types';
 
-export const JobsPageContent = () => {
-  const {
-    allRepos,
-    visibleRepos,
-    loading: workspaceLoading,
-    data: cachedData,
-  } = useContext(WorkspaceContext);
-  const [searchValue, setSearchValue] = useQueryPersistedState<string>({
-    queryKey: 'search',
-    defaults: {search: ''},
-  });
+const FILTER_FIELDS = ['jobs', 'tags', 'codeLocations'] as const;
 
+export const JobsPageContent = () => {
+  const {allRepos, visibleRepos, loading, data: cachedData} = useContext(WorkspaceContext);
   const repoCount = allRepos.length;
-
-  const queryResultOverview = useQuery<OverviewJobsQuery, OverviewJobsQueryVariables>(
-    OVERVIEW_JOBS_QUERY,
-    {
-      fetchPolicy: 'network-only',
-      notifyOnNetworkStatusChange: true,
-    },
-  );
-  const {data, loading: queryLoading} = queryResultOverview;
-
-  const refreshState = useQueryRefreshAtInterval(queryResultOverview, FIFTEEN_SECONDS);
-
   // Batch up the data and bucket by repo.
   const repoBuckets = useMemo(() => {
     const cachedEntries = Object.values(cachedData).filter(
       (location): location is Extract<typeof location, {__typename: 'WorkspaceLocationEntry'}> =>
         location.__typename === 'WorkspaceLocationEntry',
     );
-    const workspaceOrError = data?.workspaceOrError;
-    const entries =
-      workspaceOrError?.__typename === 'Workspace'
-        ? workspaceOrError.locationEntries
-        : cachedEntries;
     const visibleKeys = visibleRepoKeys(visibleRepos);
-    return buildBuckets(entries).filter(({repoAddress}) =>
+    return buildBuckets(cachedEntries).filter(({repoAddress}) =>
       visibleKeys.has(repoAddressAsHumanString(repoAddress)),
     );
-  }, [cachedData, data, visibleRepos]);
-
-  const loading = !data && workspaceLoading;
+  }, [cachedData, visibleRepos]);
+
+  const allJobs = useMemo(() => repoBuckets.flatMap((bucket) => bucket.jobs), [repoBuckets]);
+  const allTags = useTagsForObjects(allJobs, (job) => job.tags);
+
+  const {state: _state, setters} = useQueryPersistedFilterState<{
+    jobs: string[];
+    tags: Tag[];
+    codeLocations: RepoAddress[];
+  }>(FILTER_FIELDS);
+
+  const state = useMemo(() => {
+    return {
+      ..._state,
+      codeLocations: _state.codeLocations.map(({name, location}) =>
+        buildRepoAddress(name, location),
+      ),
+    };
+  }, [_state]);
```

Comment:
Isn't `codeLocations` already `RepoAddress[]`?

Reply:
It's grabbing it from the URL with JSON.parse so it ends up being a different reference from the one `buildRepoAddress` returns (since it memoizes) and we rely on referential equality in useStaticSetFilter so I have to make the references in the state align with the references we pass in .